src/model/application/interaction/application_command.rs

```diff
@@ -242,6 +242,11 @@ impl<'de> Deserialize<'de> for CommandInteraction {
     fn deserialize<D: Deserializer<'de>>(deserializer: D) -> StdResult<Self, D::Error> {
         let mut interaction = Self::deserialize(deserializer)?; // calls #[serde(remote)]-generated inherent method
         if let Some(guild_id) = interaction.guild_id {
+            if let Some(member) = &mut interaction.member {
+                member.guild_id = guild_id;
+                // If `member` is present, `user` wasn't sent and is still filled with default data
+                interaction.user = member.user.clone();
+            }
             interaction.data.resolved.roles.values_mut().for_each(|r| r.guild_id = guild_id);
```

Comment:
After taking a closer look at #2287, I noticed this line was missing from the implementations for `ComponentInteraction` and `ModelInteraction`, even though they both previously called `add_guild_id_to_resolved`. Was this intentional on your part?

Reply:
Yes, that was actually intentional. add_guild_id_to_resolved did two things:
- insert guild_id into Member
- insert guild_id into CommandData::resolved::roles

The latter only applies to CommandInteraction's so I just inlined the former into ComponentInteraction and ModalInteraction.